### PR TITLE
Archi v0.2

### PR DESCRIPTION
**Pull request by Saverio976**

### CODE REVIEW

File: include/IGameLogic.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#include "IEntity.hpp"
+#include "IEventManager.hpp"
+#include "ISystem.hpp"
+
+namespace Arcade {
+    namespace Game {
+        class IGameLogic : ISystem {
+          public:
+            virtual ~IGameLogic() = default;
+            virtual void run(
```
Comment (guillaumeAbel) [inline include/IGameLogic.hpp]: need delta Time

File: include/IEntity.hpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#pragma once
+
+#include <vector>
+#include "IComponent.hpp"
+
+namespace Arcade {
+    class IEntity {
+      public:
+        virtual ~IEntity() = 0;
+        virtual std::vector<IComponent *> &getComponents() = 0;
```
Comment (guillaumeAbel) [inline include/IEntity.hpp]: const

File: include/IEventManager.hpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    //     KEY_F11_PRESSED,
+    //     KEY_F12_PRESSED,
+    //     WINDOW_CLOSE,
+    //     WINDOW_RESIZE,
+    //     WINDOW_MINIMIZE
+    // };
+
+    class IEventManager {
+      public:
+        virtual ~IEventManager() = default;
```
Comment (guillaumeAbel) [inline include/IEventManager.hpp]: = 0

File: include/IGameLogic.hpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#include <vector>
+#include "IEntity.hpp"
+#include "IEventManager.hpp"
+#include "ISystem.hpp"
+
+namespace Arcade {
+    namespace Game {
+        class IGameLogic : ISystem {
+          public:
+            virtual ~IGameLogic() = default;
```
Comment (guillaumeAbel) [inline include/IGameLogic.hpp]: = 0

File: include/IGameLogic.hpp [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            virtual void run(
+            std::vector<IEntity *> &, const IEventManager &) = 0;
```
Comment (guillaumeAbel) [inline include/IGameLogic.hpp]: override

File: include/IMusic.hpp [hunk 6]
```diff
@@ -5,28 +5,27 @@
...
+#include "IComponent.hpp"
 
 namespace Arcade {
-    class IMusic : public IRenderComp {
-      public:
-        virtual ~IMusic() = default;
+    namespace Graph {
+        class IMusic : IComponent {
+          public:
+            virtual ~IMusic() = default;
```
Comment (guillaumeAbel) [inline include/IMusic.hpp]: = 0

File: include/IGameLogic.hpp (hunk 1 above)
Comment (guillaumeAbel) [inline include/IGameLogic.hpp]: Not modifying sytem instance but the entities, so it's const, and same for both childs

File: include/ISprite.hpp [hunk 7]
```diff
@@ -5,59 +5,29 @@
...
-        float x;
-        float y;
-    };
-
-    class ISprite : public IRenderComp {
-      public:
-        virtual ~ISprite() = default;
+        class ISprite : public IComponent {
+          public:
+            virtual ~ISprite() = default;
```
Comment (guillaumeAbel) [inline include/ISprite.hpp]: = 0

File: include/ISystemManager.hpp [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+#include <string>
+#include <vector>
+#include "IEntity.hpp"
+#include "ISystem.hpp"
+
+namespace Arcade {
+    class ISystemManager {
+      public:
+        virtual ~ISystemManager() = default;
```
Comment (guillaumeAbel) [inline include/ISystemManager.hpp]: 2 space indentation

File: include/IText.hpp [hunk 9]
```diff
@@ -5,37 +5,34 @@
...
 namespace Arcade {
-    class IText : public IRenderComp {
-      public:
-        virtual ~IText() = default;
-
-        virtual const std::string &getFontPath() const = 0;
+    namespace Graph {
+        class IText : IComponent {
+          public:
+            virtual ~IText() = default;
```
Comment (guillaumeAbel) [inline include/IText.hpp]: = 0

File: tests/dawa.cpp
Comment (guillaumeAbel) [inline tests/dawa.cpp:1]: Remove or rename

**Review by guillaumeAbel — CHANGES_REQUESTED** [review]

File: include/IEntity.hpp (hunk 2 above)
Comment (Saverio976) [reply include/IEntity.hpp]: where const ?

**Comment by Saverio976** [comment]
uml

File: include/IEntity.hpp [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    namespace ECS {
+        class IEntity {
+            public:
+                virtual ~IEntity() = default;
+                /**
+                 * @brief Get all components
+                 *
+                 * @return The list of components
+                 */
+                virtual const std::vector<IComponent *> &getComponents() = 0;
```
Comment (romainpanno) [inline include/IEntity.hpp:23]: Pourquoi faire des pointeurs, si on fait des smart ptr on peut faire des références non?

File: include/IEventManager.hpp [hunk 11]
```diff
@@ -0,0 +1,143 @@
+/*
+** EPITECH PROJECT, 2023
+** Visual Studio Live Share (Workspace)
+** File description:
+** ISystem
+*/
+
+#ifndef IUSERINPUTMANAGER_HPP_
```
Comment (romainpanno) [inline include/IEventManager.hpp:8]: Pragma once?

File: tests/test-includes.cpp [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+#include "../include/ICore.hpp"
+#include "../include/IEntity.hpp"
+#include "../include/IEventManager.hpp"
+#include "../include/IGameLogic.hpp"
+#include "../include/IGraph.hpp"
+#include "../include/IMusic.hpp"
+#include "../include/ISprite.hpp"
+#include "../include/ISystem.hpp"
+#include "../include/ISystemManager.hpp"
+#include "../include/IText.hpp"
```
Comment (romainpanno) [inline tests/test-includes.cpp:12]: Pourquoi ../ ? Si le cmake est bien fait pas besoin

Comment (romainpanno) [reply tests/test-includes.cpp:12]: Ah c'est un test

File: include/IEntity.hpp (hunk 2 above)
Comment (guillaumeAbel) [reply include/IEntity.hpp]: Missing const because its a getter

File: include/IGameLogic.hpp (hunk 1 above)
Comment (guillaumeAbel) [reply include/IGameLogic.hpp]: yes at least i think its better to let everyone to have attributes and modify it in game and graph systems

File: tests/test-includes.cpp (hunk 12 above)
Comment (Saverio976) [reply tests/test-includes.cpp:12]: Oui

File: include/IEventManager.hpp (hunk 11 above)
Comment (Saverio976) [reply include/IEventManager.hpp:8]: Smart

File: include/ArcadeStruct.hpp [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+** EPITECH PROJECT, 2023
+** Arcade
+** File description:
+** Useful struct for all the project
+*/
+
+#pragma once
+
+namespace Arcade {
+    struct Vector2f {
```
Comment (KitetsuK) [inline include/ArcadeStruct.hpp:11]: Why is z here for a vector2f ?

File: include/ICore.hpp [hunk 14]
```diff
@@ -5,17 +5,20 @@
...
 */
 
-#ifndef ICORE_HPP_
-#define ICORE_HPP_
+#pragma once
 
 namespace Arcade {
     namespace Core {
+        /**
+         * @brief The Core class
```
Comment (KitetsuK) [inline include/ICore.hpp:13]: ICore ? Because Core is the namespace so i dont know if it's good ?

File: include/IEntity.hpp [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+*/
+
+#pragma once
+
+#include <vector>
+#include "IComponent.hpp"
+
+namespace Arcade {
+    namespace ECS {
+        class IEntity {
```
Comment (KitetsuK) [inline include/IEntity.hpp:15]: No doc before IEntity ? To explain what it is

File: include/IGameLogic.hpp [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <cstddef>
+#include <vector>
+#include "IEntity.hpp"
+#include "IEventManager.hpp"
+#include "ISystem.hpp"
+
+namespace Arcade {
+    namespace Game {
+        class IGameLogic : Arcade::ECS::ISystem {
```
Comment (KitetsuK) [inline include/IGameLogic.hpp:18]: No doc for this class, i don t know if it's necessary but it could be nice

**Comment by Saverio976** [comment]
> Y a beaucoup de class qui ont pas de doc, je sais pas si c'est normal mais je préfère demander

Oui, j'ai documenté les classes avec des fonctions qui peuvent prêter à confusion

File: include/IEntity.hpp (hunk 15 above)
Comment (Saverio976) [reply include/IEntity.hpp:15]: It's an ECS, anyone who will use this, I hope understood it before the usage of this interface (at meetings that we set-up every day)

File: include/IGameLogic.hpp (hunk 16 above)
Comment (Saverio976) [reply include/IGameLogic.hpp:18]: I don't know what to say in the doc for this

File: include/ArcadeStruct.hpp (hunk 13 above)
Comment (guillaumeAbel) [reply include/ArcadeStruct.hpp:11]: For have only 1 struct to represent pos or size in 2d and 3d, if its 2d z is always 0

**Comment by Saverio976** [comment]
#7

**Review by KitetsuK — APPROVED** [review]

**Review by guillaumeAbel — APPROVED** [review]

**Pull request merged**